WordPress/gutenberg
# Add context properties to block types REST endpoint

This pull request seeks to add [block context](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-context.md) properties to the block types REST controller originally implemented in #21065. Block context documentation was created as part of #21925, but not incorporated into the existing Block Type RFC document. It has been included here alongside the necessary revisions to include the properties in the REST controller.

**Status:** The implementation should work well enough, though it may warrant being considered blocked by one or both of:

- A decision at https://github.com/WordPress/gutenberg/pull/22519#discussion_r431461183 in regards to registering block types using snake-cased `provides_context`, rather than camel-cased `providesContext`.
- A solution to provide `provides_context` as an empty _object_ `{}`, rather than incorrect array `[]`, related to the comment at https://github.com/WordPress/gutenberg/pull/21065/files#r431468417.

**Testing Instructions:**

Ensure tests pass:

```
./packages/env/bin/wp-env run composer 'composer install'
npm run test-unit-php
```

Test scenario from #21467 is helpful as well:

> Given that this pull request only ports the Post Title experimental full-site editing block, testing can be a bit cumbersome at the moment, especially if you've not yet defined templates for full-site editing mode.
> 
> My recommendation for testing is as follows:
> 
> 1. Navigate to Gutenberg > Experiments
> 2. Enable "Full Site Editing" and click "Save Changes"
> 3. Navigate to Posts > Add New
> 4. Insert a Post Title block
> 5. Observe that the block inherits the title of the post and updates if you change the title
> 6. Preview the post in a separate tab
>    - Note: If you don't have templates on your site, you may see a message "No template found". This is expected, and can be disregarded for now.
> 7. Navigate to Gutenberg > Experiments
> 8. Disable "Full Site Editing" and click "Save Changes"
> 9. In your separate preview tab, refresh the page.
> 10. Observe that the Post Title block displays the title of the current post
